### DISCUSSION TITLE
Does ANE plugin support m1 mac?

### DISCUSSION THREAD

**Discussion by sol-oka** [airsdk/Adobe-Runtime-Support#2092] [Q&A] [answered]
I have ANE plugin written by myself. It can work on Intel MAC, but can't run on m1 MAC.

**Comment by ajwfrost** [comment, marked as answer]
Is your library executable file a universal binary? i.e. having both Intel-based (x86_64) and M1/Apple-based (arm64) versions packaged into the same file?

For macOS because of the universal binary system, we don't treat Apple silicon as a different platform, which may be a bit confusing.. but if you package up just as a normal Intel macOS ANE and the binary includes the arm64 version, then it should work..

thanks

**Reply by sol-oka** [reply]
![xcode](https://user-images.githubusercontent.com/102789019/186929344-bfd88119-baa6-4152-b4d1-af51b5fe43d4.png)
I have add the arm and x86 in the compile setting.

**Reply by sol-oka** [reply]
**adt -package -target ane ane/ANEPlugins.ane aneplugins/extension.xml -swc ANEPlugins.swc -platform MacOS-x86-64 -C aneplugins/MacOS-x86-64 .** 

Do the build ANE command line have a universal binary  or ARM version?

**Reply by ajwfrost** [reply]
We'd looked at changing the ANE command line for this but in the end it's just best to keep as `MacOS-x86-64` as you have there.

The settings has `Architectures` being both, but `Build Active Architecture Only` is set to true which means it will only actually build the architecture of your machine. You can check this if you find the actual executable file and use the `file` command, or `lipo`, in a terminal window..

thanks

**Reply by sol-oka** [reply]
Thanks for your reply. I will test it again.
